operatorapi/volumes.go

```
@@ -218,3 +228,36 @@ func getPVCEventsResponse(session *models.Principal, params operator_api.GetPVCE
 	})
 	return retval, nil
 }
+
+func getTenantCSResponse(session *models.Principal, params operator_api.ListTenantCertificateSigningRequestParams) (*models.CsrElement, *models.Error) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	clientset, err := cluster.K8sClient(session.STSSessionToken)
+	if err != nil {
+		return nil, errors.ErrorWithContext(ctx, err)
+	}
+	csrName := params.Tenant + "-" + params.Namespace + "-csr"
+	csrResult, csrError := clientset.CertificatesV1().CertificateSigningRequests().Get(ctx, csrName, metav1.GetOptions{})
+	if csrError != nil {
+		panic(csrError.Error())
```

let's not use panics, just print the error and return the API with an error, better to return a model error here

---

Ok @dvaldivia make sense, thank you!